helion/autotuner/pattern_search.py

```diff
@@ -40,6 +40,9 @@ def __init__(
             max_generations: The maximum number of generations to run.
         """
         super().__init__(kernel, args)
+        max_generations_override = kernel.settings.autotune_max_generations
+        if max_generations_override is not None:
+            max_generations = max_generations_override
```

Comment:
Ignoring the value passed to the constructor doesn't seem right.

The priority should go:
1) An explicit value passed to the constructor
2) An explicit value from global settings
3) The default value (autotuner specfic)